documentcloud/documents/tests/test_oembed.py

```diff
@@ -0,0 +1,335 @@
+# Django
+from django.conf import settings
+from django.test import RequestFactory, TestCase
+
+# Standard Library
+from unittest import mock
+
+# DocumentCloud
+from documentcloud.documents.models import Document
+from documentcloud.documents.oembed import DocumentOEmbed, NoteOEmbed, PageOEmbed
+from documentcloud.oembed.utils import Query
+
+
+class DocumentOEmbedTest(TestCase):
+    # pylint: disable=too-many-instance-attributes
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = mock.MagicMock()
+        self.document = mock.MagicMock(spec=Document)
+        self.document.title = "Test Document"
+        self.document.aspect_ratio = 1.5
+        self.document.slug = "test-document"
+        self.document.pk = 123
+        self.document.get_absolute_url.return_value = (
+            f"/documents/{self.document.pk}-{self.document.slug}/"
+        )
+
+        # Mock the Document manager
+        self.document_queryset = mock.MagicMock()
+        self.document_queryset.get_viewable.return_value = self.document_queryset
+        self.document_queryset.get.return_value = self.document
+        self.original_objects = Document.objects
+        Document.objects = self.document_queryset
+
+        # Create a DocumentOEmbed instance
+        self.document_oembed = DocumentOEmbed()
+
+        # Mock get_object_or_404
+        self.get_object_patcher = mock.patch(
+            "documentcloud.documents.oembed.get_object_or_404",
+            return_value=self.document,
+        )
+        self.mock_get_object = self.get_object_patcher.start()
+
+    def tearDown(self):
+        # Restore original objects
+        Document.objects = self.original_objects
+        # Stop all patches
+        self.get_object_patcher.stop()
+
+    def test_document_oembed_response(self):
+        """Test the response method of DocumentOEmbed"""
+        request = self.factory.get("/oembed/")
+        request.user = self.user
+        query = Query("responsive=1")
+
+        response = self.document_oembed.response(
+            request, query, max_width=600, max_height=None, pk=123
+        )
+
+        # Check response properties
+        self.assertEqual(response["version"], "1.0")
```

Comment:
Since we are using `pytest` you can just write asserts instead of using the assert methods, and it will handle them nicely - see other tests in DocumentCloud.  This can be written as:
`assert response["version"] == "1.0"`